connection.ts

```diff
@@ -522,10 +529,47 @@ export class Connection {
   }
 
   async query(query: Query): Promise<QueryResult> {
-    if (query.args.length === 0) {
-      return await this._simpleQuery(query);
+    const queueItem: QueryQueueItem = {
+      query: query,
+      callback: undefined,
+      started: false,
+    };
+
+    const queryPromise = new Promise<QueryResult>((resolve, reject) => {
+      queueItem.callback = (
```

Comment:
If there's a better way to return a promise that can later be resolved that would be great!